modules/kubernetes-addons/aws-ebs-csi-driver/main.tf

```
@@ -19,11 +22,11 @@ resource "aws_eks_addon" "aws_ebs_csi_driver" {
 module "irsa_addon" {
   source = "../../../modules/irsa"
 
-  count = local.create_irsa ? 1 : 0
+  count = local.create_irsa && var.enable_amazon_eks_aws_ebs_csi_driver && !var.enable_self_managed_aws_ebs_csi_driver ? 1 : 0
```

this should stay as it was - IRSA is valid with either route

---

to deploy the addon in self_managed mode i use the terraform module "helm_addon" which contains [a block for IRSA management](https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/896/files#diff-a3e7919ec85e9b3b1439fe0505889c99852929bcb828868f7545c2bb5870a0f6R82-R88). so i used it and disabled this resource when we are in self managed mode

---

ah right, my apologies. Then I think what you want is

```suggestion
  count = local.create_irsa && !var.enable_self_managed_aws_ebs_csi_driver ? 1 : 0
```